synapse/events/third_party_rules.py

```diff
@@ -77,29 +194,31 @@ async def check_event_allowed(
         # the hashes and signatures.
         event.freeze()
 
-        return await self.third_party_rules.check_event_allowed(event, state_events)
+        for callback in self._check_event_allowed_callbacks:
+            res = await callback(event, state_events)
+            # Return if the event shouldn't be allowed or if the module came up with a
+            # replacement content for the event.
+            # TODO: is this really the right place to let modules replace the event's
+            #  content (or should it happen in another callback, or as a second return
+            #  argument)?
+            if not res or isinstance(res, dict):
+                return res
+
+        return True
 
     async def on_create_room(
         self, requester: Requester, config: dict, is_requester_admin: bool
-    ) -> bool:
-        """Intercept requests to create room to allow, deny or update the
-        request config.
+    ) -> None:
+        """Intercept requests to create room to maybe deny it (via an exception) or
+        update the request config.
 
         Args:
             requester
             config: The creation config from the client.
             is_requester_admin: If the requester is an admin
-
-        Returns:
-            Whether room creation is allowed or denied.
         """
-
-        if self.third_party_rules is None:
-            return True
-
-        return await self.third_party_rules.on_create_room(
-            requester, config, is_requester_admin
-        )
+        for callback in self._on_create_room_callbacks:
+            await callback(requester, config, is_requester_admin)
```

Comment:
Should we add some error handling so that one broken module doesn't prevent others from executing this method?